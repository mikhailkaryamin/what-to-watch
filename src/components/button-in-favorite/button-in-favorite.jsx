import React from 'react';
import {connect} from 'react-redux';

import {
  func,
  number,
} from 'prop-types';

import {Operation as FavoriteOperation} from '../../reducer/favorite/favorite.js';

const ButtonInFavorite = (props) => {
  const {
    id,
    setFavorite,
  } = props;

  return (
    <button
      className="btn movie-card__button btn--list"
      type="button"
      onClick={() => setFavorite(id)}
    >

      <svg width="19" height="20" viewBox="0 0 19 20">
        <use xlinkHref="#add"></use>
      </svg>

      <span>
        My list
      </span>
    </button>
  );
};

ButtonInFavorite.propTypes = {
  id: number.isRequired,
  setFavorite: func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  setFavorite: (id) => {
    dispatch(FavoriteOperation.setFavoriteFilm(id));
  }
});

export {ButtonInFavorite};

export default connect(null, mapDispatchToProps)(ButtonInFavorite);
