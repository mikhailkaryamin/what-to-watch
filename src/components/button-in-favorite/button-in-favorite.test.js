import React from 'react';
import renderer from 'react-test-renderer';

import {ButtonInFavorite} from './button-in-favorite.jsx';

describe(`Button in favorite component`, () => {
  const FILM_CARD = {
    backgroundImage: `img/aviator.jpg`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
    Aliquam a justo elit. Nulla vitae hendrerit dolor. Sed luctus massa lectus.`,
    director: `Arthur Agee`,
    genre: `Comedy`,
    id: 1,
    isFavorite: false,
    name: `Devin Albert`,
    posterImage: `img/aviator.jpg`,
    previewImage: `img/aviator.jpg`,
    previewVideoLink: `link`,
    rating: 3,
    released: 2015,
    runTime: 100,
    scoreCount: 300,
    starring: [`Arthur Agee`, `Robert Agnew`, `Alan Aisenberg`],
    video: `some/link`,
  };

  test(`should render button in favorite`, () => {

    const wrapper = renderer
      .create(
          <ButtonInFavorite
            film={FILM_CARD}
            place={``}
            setFavorite={() => {}}
          />
      ).toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
