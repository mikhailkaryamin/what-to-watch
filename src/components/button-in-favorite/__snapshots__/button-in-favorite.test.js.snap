// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button in favorite component should render button in favorite 1`] = `
<button
  className="btn movie-card__button btn--list"
  onClick={[Function]}
  type="button"
>
  <svg
    height="20"
    viewBox="0 0 19 20"
    width="19"
  >
    <use
      xlinkHref="#add"
    />
  </svg>
  <span>
    My list
  </span>
</button>
`;
